lib/compressTextureCoordinates.js

```diff
@@ -0,0 +1,172 @@
+            var primitive = texCoordAccessors[texCoordAccessorId];
+            if (canCompress(accessor, primitive)) {
+                // Compress texture coordinates to a single float
+                var bufferViewId = accessor.bufferView;
+                var bufferView = bufferViews[bufferViewId];
+                var bufferId = bufferView.buffer;
+                var buffer = buffers[bufferId];
+                var source = buffer.extras._pipeline.source;
+                var byteStride = getAccessorByteStride(accessor);
+                var byteOffset = accessor.byteOffset + bufferView.byteOffset;
+                var componentType = accessor.componentType;
+                var newComponentType = WebGLConstants.FLOAT;
+                var componentByteLength = byteLengthForComponentType(componentType);
+                var numElements = accessor.count;
+                var numComponents = numberOfComponentsForType(accessor.type);
+                for (var j = 0; j < numElements; j++) {
+                    var elementArray = [];
+                    for (var k = 0; k < numComponents; k++) {
+                        var element = readBufferComponentType(source, componentType, byteOffset + componentByteLength * k);
+                        elementArray.push(element);
+                    }
+                    var texCoord = Cartesian2.fromArray(elementArray);
+                    var compressed = AttributeCompression.compressTextureCoordinates(texCoord);
+                    writeBufferComponentType(source, newComponentType, compressed, byteOffset);
+                    byteOffset += byteStride;
+                }
+                accessor.byteStride = byteStride;
+                accessor.componentType = newComponentType;
+                accessor.type = 'SCALAR';
+                var minMax = findAccessorMinMax(gltf, accessor);
+                accessor.min = minMax.min;
+                accessor.max = minMax.max;
+                var materialId = primitive.material;
+                var material = materials[materialId];
+                var techniqueId = material.technique;
+                var attributeVariableName = patchedTechniques[techniqueId];
+                if (!defined(patchedTechniques[techniqueId])) {
+                    attributeVariableName = patchTechnique(gltf, techniqueId);
+                    patchedTechniques[techniqueId] = attributeVariableName;
+                }
+                var technique = techniques[techniqueId];
+                var programId = technique.program;
+                if (!defined(patchedPrograms[programId])) {
+                    patchProgram(gltf, programId, attributeVariableName);
+                    patchedPrograms[programId] = true;
+                }
+            }
+        }
+    }
+    uninterleaveAndPackBuffers(gltf);
+}
+
+function canCompress(accessor) {
+    return accessor.type === 'VEC2';
+}
+
+function getTextureCoordinateAccessors(gltf) {
```

Comment:
Actually, it could be more elegant and faster (by avoiding the intermediate data structure) to have this new function take a function that is called per accessor when it is discovered.  In general, I think helpers like this that help walk the glTF data structures could simplify a lot of code in gltf-pipeline.